docs/snapshotting/snapshot-support.md

```diff
@@ -0,0 +1,280 @@
+            \"snapshot_path\": \"/path/to/snapshot_file\",
+	        \"mem_file_path\": \"/path/to/mem_file\",
+            \"version\": \"2\"
+    }"
+```
+
+Details about the required and optional fields can be found in the
+[swagger definition](../../src/api_server/swagger/firecracker.yaml).
+
+**Prerequisites**: It can be called only when the microVM is `Paused`.
+**Effects**:
+- _on success_:
+  - The file indicated by `/path/to/snapshot_file` contains the devices' model state
+    and emulation state. The one indicated by `/path/to/mem_file` contains a full copy
+    of the guest memory.
+  - The block devices contents are flushed to their backing files. At this point, these
+    should be backed up externally by the user. The generated snapshot files are
+    immediately available to be used (current process releases ownership).
+  - The snapshot creation resets then the dirtied page bitmap and marks all pages clean
+    (from a diff snapshot point of view).
+
+If a `version` is specified, the new snapshot is saved at that version, otherwise
+it will be saved at the same version of the running Firecracker. The version is only
+used for the microVM state file as it contains internal state structures for device
+emulation, vCPUs and others that can change their format from a Firecracker version
+to another. Versioning is not supported for block devices and memory files. The separate
+block device file components of the snapshot have to be handled by the user.
+
+- _on failure_: no side-effects.
+
+### Creating diff snapshots
+
+For creating a diff snapshot, you should use the same API command, but with
+`snapshot_type` field set to `Diff`.
+*Note*: If omitted from the request body, `snapshot_type` default value is `Full`.
+
+```bash
+curl --unix-socket /tmp/firecracker.socket -i \
+    -X PUT "http://localhost/snapshot/create" \
+    -H  "accept: application/json" \
+    -H  "Content-Type: application/json" \
+    -d "{
+            \"snapshot_type\": \"Diff\",
+            \"snapshot_path\": \"/path/to/snapshot_file\",
+	        \"mem_file_path\": \"/path/to/mem_file\",
+            \"version\": \"2\"
+    }"
+```
+
+**Prerequisites**: It can be called only when the microVM is `Paused`.
+**Effects**:
+- _on success_:
+The file indicated by `/path/to/snapshot_file` contains the devices' model state and
+emulation state, same as when creating a full snapshot. The one indicated by
+`/path/to/mem_file` contains this time a **diff copy** of the guest memory - the diff
+is consisted of the memory pages which have been dirtied since the last snapshot creation
+or since the creation of the microVM, whichever of these events was the most recent.
+All the other effects mentioned in the **Effects** section from `## Creating full snapshots`
+apply here.
+- _on failure_: no side-effects.
+
+To be able to do any diff snapshots, `track_dirty_pages` field should be set to `true`
+when configuring the `/machine-config` resource. Example:
```

Comment:
This is a prerequisite.

Reply:
Also you would need to detail both the boot/load setting of track_dirty_pages.

Reply:
Done.